### PR TITLE
CI: use windows-2022

### PR DESCRIPTION
**Pull request by pheiduck**
WIP PR to get the CI ready for Windows 2022

### CODE REVIEW

**Comment by pheiduck** [comment]
ping @k-takata

**Comment by k-takata** [comment]
Oh, mistake.

**Comment by pheiduck** [comment]
LGTM now, only some tests failing.
No CI related issues.

**Comment by k-takata** [comment]
As I wrote in the previous PR, Python 2 is not tested on 32-bit Vim:
https://github.com/vim/vim/runs/6875469662?check_suite_focus=true#step:11:102

**Comment by pheiduck** [comment]
> As I wrote in the previous PR, Python 2 is not tested on 32-bit Vim: https://github.com/vim/vim/runs/6875469662?check_suite_focus=true#step:11:102

Than we need to install python2-x86_32 --version=2.7.8

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -423,14 +424,17 @@ jobs:
         run: |
           git config --global core.autocrlf input
           echo "VCVARSALL=$(vswhere -products \* -latest -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
-          python_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON_VER_DOT}/InstallPath/@")
+          choco install python2 -ForceX86
```
Comment (k-takata) [inline .github/workflows/ci.yml]:
```suggestion
          if [ "${{ matrix.arch }}" = "x86" ]; then
            choco install python2 --forcex86
          else
            choco install python2
          fi
```

**Comment by pheiduck** [comment]
🎉 all Windows CIs are ✅

**Pull request closed without merging**